### DISCUSSION TITLE
Support "app" based trust management with trust.d

### DISCUSSION THREAD

**Discussion by jw3** [ctc-oss/fapolicy-analyzer#281] [Developers]
Given the addition of trust.d support we could move to a design that is aware of app bundles for trust and manage those bundles in their own entry in the trust directory.

Currently we flatten everything to the fapolicyd.trust file, which is simple for writing, but will eventually create some difficulty in reasoning about what has been added to that file.

**Comment by radosroka** [comment]
With trust.d you can create multiple trust files.
The name of the trust file holds the information where it came from.

However, fapolicyd daemon does not hold the file name any longer after reading the file.
All the entries from trust.d have `file` set as a source in trustdb,

**Comment by jw3** [comment]
> All the entries from trust.d have file set as a source in trustdb,

Yep, this is a good point.  Right now we only parse the lmdb for a source of trust.  To achieve the modularity and provenance I was getting at in the original post we will either have to track "apps" in application storage somewhere (XDG most likely) or will have to start parsing the file trust sources ourselves.

Organizing our "apps" in trust.d let's us avoid having to check sync between our state and the trust.d state (eg. check if someone edited those by hand outside our app).  So that would seem to be the place to start.
